### PR TITLE
Use GPU and batch the training sample if requested

### LINKED ISSUE #5: Make flow-training code device agnostic
i.e. use GPU if the computer has one and the user requests it to be used.

### PR DESCRIPTION
**Pull request by rjhunter999**
Changes the shared `train_flow` function to put the sample tensors and the flow on the GPU if requested.

Provides `--train-on-gpu` flag for `ip_correction_flow.py` to enable this on demand, while `flow_to_lhcb_ip_sim.py` now does this by default. 

Also enables batching of the training data sample, but this is switched off by default as it shows no performance gain. 

Using the GPU currently shows no performance gain (runtime of `flow_to_lhcb_ip_sim.py` is unchanged by toggling the GPU usage off or on on my machine), which probably shows that the sample size and computational complexity isn't sufficient for a speed up. Further studies would be required to speed up the training e.g. into the hyperparameters, required training sample size and number of iterations, learning rate etc. 

Closes #5.

### CODE REVIEW

**Comment by rjhunter999** [comment]
Code looks good; changes are short. Merging!
